samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManagerFactory.scala

```diff
@@ -62,16 +64,23 @@ class KafkaCheckpointManagerFactory extends CheckpointManagerFactory with Loggin
     val jobName = config.getName.getOrElse(throw new SamzaException("Missing job name in configs"))
     val jobId = config.getJobId.getOrElse("1")
 
-    val systemName = config
-      .getCheckpointSystem
-      .getOrElse(throw new SamzaException("no system defined for Kafka's checkpoint manager."))
+    val (systemName: String, systemFactory : SystemFactory) =  org.apache.samza.util.Util.getCheckpointSystemStreamAndFactory(config)
 
     val producerConfig = config.getKafkaSystemProducerConfig(
       systemName,
       clientId,
       INJECTED_PRODUCER_PROPERTIES)
-    val connectProducer = () => {
-      new KafkaProducer[Array[Byte], Array[Byte]](producerConfig.getProducerProperties)
+
+    val systemProducerF = () => {
+      systemFactory.getProducer(systemName, config, new NoOpMetricsRegistry())
+    }
+
+    val systemConsumerF = () => {
+      systemFactory.getConsumer(systemName, config, new NoOpMetricsRegistry())
```

Comment:
Can we share the same NoOpMetricsRegistry?

Reply:
Done.